compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -128,6 +128,48 @@ if (cg->profiledPointersRequireRelocation() && secondChild->getOpCodeValue() ==
    }
 #endif
 
+   if (secondChild->getOpCode().isLoadConst())
+      {
+      int64_t secondChildValue = is64bit ? secondChild->getLongInt() : secondChild->getInt();
+      if ((cc == TR::CC_EQ || cc == TR::CC_NE)
+            && secondChildValue == 0)
+         {
+         TR::InstOpCode::Mnemonic op;
+         bool isEqualCmp = TR::ILOpCode::isEqualCmp(node->getOpCode().getOpCodeValue());
+         bool isNotEqualCmp = TR::ILOpCode::isNotEqualCmp(node->getOpCode().getOpCodeValue());
+         TR_ASSERT_FATAL(isEqualCmp || isNotEqualCmp, "The node has to be isEqualCmp or isNotEqualCmp");
+         if (isEqualCmp)
```

Comment:
I think you can just check with `if (cc == TR::CC_EQ)` instead of introducing `isEqualCmp` and `isNotEqualCmp` here.
@0xdaryl may have a different opinion.

Reply:
@knn-k It was suggested by Daryl in the earlier discussion in this thread.

Reply:
Yes, I know.  But that was in `generateCbzOrCbnzInstruction()` which didn't have the `cc` in its arguments.

Reply:
Okay. I will wait for the @0xdaryl feedback. 
The assert is redundant also, as I have already checked if cc is CC_EQ or CC_NE.

Reply:
@0xdaryl, Can I have your feedback please.

Reply:
@knn-k  is correct.  If the condition code is already passed in then you can just use that.  I believe my comment earlier was in a function where it wasn't.

Reply:
Condition is changed as per discussion and I have also removed the `TR_ASSERT_FATAL` as it is redundant.